hack/test-go.sh

```diff
@@ -7,4 +7,4 @@ if [[ -n ${ARTIFACT_DIR:-} ]]; then
   JUNIT_ARG="--junitfile=$ARTIFACT_DIR/junit.xml"
 fi
 
-gotestsum $JUNIT_ARG -- ./... -race ${TESTFLAGS:-}
+gotestsum $JUNIT_ARG -- ${PACKAGES:-"./..."} -race ${TESTFLAGS:-}
```

Comment:
The common approach to not run tests by default is to use build tags: https://golang.org/pkg/go/build/#hdr-Build_Constraints

This has the advantage that things like this are not needed and ppl can still run `go test ./...`

Reply:
If someone's running `go test ./...` I want them to have the full suite. Build tags are pretty heavily linked with OS/ARCH and that seems like a misuse to me IMO

Reply:
Arch-specifics and tests are the main use case for build tags, just Google it, you will find plenty of examples for this. No one wants tests that require special setup and won't work by default to run when they run a simple go test. /..... As a sideffrct, this also avoids having to adjust all existing test commands